From c72599a7bd46c22d296c59dbd13f138503350d2f Mon Sep 17 00:00:00 2001
Message-Id: <c72599a7bd46c22d296c59dbd13f138503350d2f.1367382591.git.dvhart@linux.intel.com>
From: Darren Hart <dvhart@linux.intel.com>
Date: Tue, 30 Apr 2013 21:11:17 -0700
Subject: [PATCH 1/4] configure.ac: Add option to disable configuring the BIOS
 test

When cross-compiling, AC_CHECK_FILE aborts configure. Provide a means to
avoid even attempting to configure the BIOS test.

Signed-off-by: Darren Hart <dvhart@linux.intel.com>
---
 configure.ac |   10 +++++++---
 1 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index efa0e54..524060c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -66,9 +66,13 @@ AC_SUBST([ARCH])
 AC_SUBST([MACHINE_TYPE_NAME])
 
 # test bios
-AC_CHECK_FILE([/usr/lib/qemu-bios],        [BIOS=/usr/lib/qemu-bios])
-AC_CHECK_FILE([/usr/share/qemu-ovmf/bios], [BIOS=/usr/share/qemu-ovmf/bios])
-AC_SUBST([BIOS])
+AC_ARG_ENABLE(biostest, AS_HELP_STRING([--disable-biostest], [disable building BIOS test]))
+if test "x$enable_biostest" != "xno"; then
+	AC_CHECK_FILE([/usr/lib/qemu-bios],        [BIOS=/usr/lib/qemu-bios])
+	AC_CHECK_FILE([/usr/share/qemu-ovmf/bios], [BIOS=/usr/share/qemu-ovmf/bios])
+	AC_SUBST([BIOS])
+fi
+AM_CONDITIONAL(ENABLE_BIOSTEST, [test "x$enable_biostest" = "xyes"])
 
 # ------------------------------------------------------------------------------
 dnl Compile EFI stuff is so tricky that it's probably better to check for the
-- 
1.7.5.4

